pkg/controller/storagecluster/storagecluster.go

```diff
@@ -622,6 +623,25 @@ func (c *Controller) createOrUpdateDeprecatedCRD() error {
 	return nil
 }
 
+func (c *Controller) miscCleanUp(cluster *corev1.StorageCluster) error {
+	// cleanup  depricated -T dmthin
+	if _, miscExists := cluster.Annotations[pxutil.AnnotationMiscArgs]; miscExists {
+		// remove '-T dmthin' from misc args if it exists.  It would have been added due to a bug
+		miscAnnotations := cluster.Annotations[pxutil.AnnotationMiscArgs]
+		dmThinStr := `-T *dmthin`
+		if dmExists, err := regexp.MatchString(dmThinStr, miscAnnotations); err == nil {
+			rexp := regexp.MustCompile(dmThinStr)
```

Comment:
Could you please move this local variable into global's?
* note, regexp-objects are immutable, and traditionally "expensive to create" -- so always a good idea to keep them defined as "global variables"

Reply:
uh.. also, you're actually creating the same "heavy" regex-object _twice_  (first implicitly on line 632, then explicitly on line 633)

Recommend rewriting as:
```go
// globals...
var myMagicalRexp = regexp.MustCompile(`-T *dmthin`)

func (c *Controller) miscCleanUp(cluster *corev1.StorageCluster) error {
...
   	m := myMagicalRexp.FindStringSubmatch(miscAnnotations)
        if len(m) > 0 {
           // regex matched...

```